### PR TITLE
Bugfix/nw23001440/fix IF condition in case of AND returning "false" and OR returning "true"

### PR DESCRIPTION
**Pull request by cosentino-smeup**
## Description
In this branch has been fixed the problem of the IF condition with an AND returning "false" and a OR returning "true".
There was a test which returned "false" as result of the IF statement without considering the OR conditions.

Include a summary of the change.

Related to 
- A02_P12

## Checklist:
- [X] There are tests regarding this feature
- [X] The code follows the Kotlin conventions (run `./gradlew ktlintCheck`)
- [X] The code passes all tests (run `./gradlew check`)
- [ ] There is a specific documentation in the `docs` directory

### CODE REVIEW

File: rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/evaluation/SmeupInterpreterTest.kt [hunk 1]
```diff
@@ -119,4 +119,10 @@ open class SmeupInterpreterTest : AbstractTest() {
         val expected = listOf("999-9999", "A90_A4(        ) A90_A5(RPG DEPT)", "A90_A4(        ) A90_A5(RPG DEPT)")
         assertEquals(expected, "smeup/T10_A90".outputOf())
     }
+
+    @Test
+    fun executeA02_P12() {
```
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/evaluation/SmeupInterpreterTest.kt:124]: Wrong name, the naming convention is `Txx_Ayy` with `xx` program id and `yy` section id

File: rpgJavaInterpreter-core/src/test/resources/smeup/A02_P12.rpgle
Comment (lanarimarco) [inline rpgJavaInterpreter-core/src/test/resources/smeup/A02_P12.rpgle:1]: fix program name

**Review by lanarimarco — CHANGES_REQUESTED** [review]
Fix program name

File: rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/evaluation/SmeupInterpreterTest.kt (hunk 1 above)
Comment (cosentino-smeup) [reply rpgJavaInterpreter-core/src/test/kotlin/com/smeup/rpgparser/evaluation/SmeupInterpreterTest.kt]: renamed

File: rpgJavaInterpreter-core/src/test/resources/smeup/A02_P12.rpgle
Comment (cosentino-smeup) [reply rpgJavaInterpreter-core/src/test/resources/smeup/A02_P12.rpgle:1]: fixed

**Review by lanarimarco — APPROVED** [review]
Ok

**Comment by lanarimarco** [comment]
You need to resolve conflicts too.

**Review by lanarimarco — APPROVED** [review]
Ok

**Pull request merged**